lib/ansible/modules/cloud/amazon/elasticache.py

```diff
@@ -218,21 +208,27 @@ def create(self):
             else:
                 msg = "'%s' is currently deleting. Cannot create."
                 self.module.fail_json(msg=msg % self.name)
+        if self.cache_port is None:
```

Comment:
This is good, but can you update the docs for `cache_port`?

The other way is to allow AWS to set the defaults by not passing `Port` if not set. Typically we do this by constructing a dict of arguments that we'll always pass, and then adding to the dict things that are set

```
kwargs = dict(CacheClusterId=self.name,
              NumCacheNodes=self.num_nodes,
              CacheNodeType=self.node_type,
              Engine=self.engine,
              EngineVersion=self.cache_engine_version,
              CacheSecurityGroupNames=self.cache_security_groups,
              SecurityGroupIds=self.security_group_ids,
              CacheParameterGroupName=self.cache_parameter_group,
              CacheSubnetGroupName=self.cache_subnet_group,
              PreferredAvailabilityZone=self.zone)
if self.cache_port is not None:
    kwargs['Port'] = self.cache_port
```